pwiz_tools/Skyline/Model/DocSettings/SrmSettings.cs

```diff
@@ -2594,6 +2594,8 @@ public SrmSettingsDiff(SrmSettings settingsOld, SrmSettings settingsNew)
                 }
                 uniquenessConstraintChange = uniquenessConstraintChange || !newPep.DigestSettings.Equals(oldPep.DigestSettings);
             }
+            // Don't try to update peptide uniqueness if the libraries have not yet been loaded
+            uniquenessConstraintChange = uniquenessConstraintChange && newPep.Libraries.IsLoaded;

```

Comment:
maybe (or am I showing my C++ roots?)
uniquenessConstraintChange &= newPep.Libraries.IsLoaded;

Reply:
Yeah. Bad C++ habits die hard, like using bitwise & on booleans. ReSharper would be well within its rights to warn you about this.

Reply:
Seems like more concise syntax - why is it problematic?

Reply:
The trend from C++ to Java/C# is to make operations more type-explicit, and & is a bitwise operator, not a boolean operator. Many of the "cleaver" conveniences of C++ have died in Java/C# and thankfully even some of the stylistically  cleaver syntax to avoid C++ pitfalls have also become unnecessary, e.g. if (null == myVal) to avoid assignment within an if statement. I applaud the Java developers who initiated this change, which makes me not very appreciative of C++ cleverness.

Interesting discussion here:

https://stackoverflow.com/questions/6346001/why-are-there-no-or-operators-in-c

Even the person with the top response who says you can use &= (instead of &&=) is not a big fan.

Reply:
I believe the line of code that Brian originally commented on is very different than the one we are discussing now.

I was shocked to see that my initial checkin had this completely nonsensical line of code in it:
`bool uniquenessConstraintChange = uniquenessConstraintChange = !Equals(newPep.Filter.PeptideUniqueness, oldPep.Filter.PeptideUniqueness);`

And I was like: "[Good catch, Brian] that's a weird line of code for me to have written..."

There's no coding style that would decide between "&=" or "= ... && "; both would be acceptable.

Reply:
So, on we go. Yes, I saw that change, and it was clearly a good one. Surprised that ReSharper wouldn't warn about the redundant assignment in the original.

Reply:
I think C++ is IMO even more type specific than C# and Java. Especially early Java and C#. Both latter languages had a lot of revisions to make them more type explicit. For example, all the reference types derive from object, there's ArrayList, and `*Collection`s in a bunch of WinForms controls that return objects instead of base types. But C++ definitely takes the cake for confusing paradigms (template metaprogramming, anyone?) and baffling and absurdly verbose compile errors. But hey, _if you do it right_, you get [zero-cost abstraction](https://joshpeterson.github.io/a-zero-cost-abstraction) (unless you need to turn off inlining and optimizations...then it's a big cost :) ).

Reply:
The stackoverflow poster has a point, though:
a = a && b can also be if (a) a = b
a = a || b can also be if (!a) a = b
And you get the ability to set a breakpoint in the deal!

Reply:
One reason to use bitwise operators for bools is if performance is a concern (e.g. inside a deep deep computational loop) and short-circuiting isn't needed. With bitwise you can avoid branches.